### PR TITLE
Lesson03

### PR DESCRIPTION
**Pull request by ann-kolosova**
Lesson03_Task01
- list sorted by age and sex fields
- list updated without first two elements and last two elements
- total numbers of "female" and "male" is calculated and printed

### CODE REVIEW

File: Lesson03_Task01.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+    ('Hannah', 22, 200, 'Williams', 'female'),
+    ('Ivan', 29, 160, 'Brown', 'male'),
+    ('Julie', 55, 90, 'Clark', 'female'),
+    ('Kenny', 38, 140, 'Harris', 'male'),
+    ('Luna', 55, 170, 'Smith', 'female'),
+    ('Mike', 55, 55, 'Johnson', 'male'),
+]
+
+updated_people = people[2:-2]
+updated_people.sort(key=itemgetter(1, 4))
```
Comment (antohaUa) [inline Lesson03_Task01.py:74]: Here you miss operations order(
First we sort then we cut. Result will be totally different .

File: Lesson03_Task01.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+import itertools
+from operator import itemgetter
+
+SEP_NUM = 17
+print('-' * SEP_NUM)
+print(f"| {'sex':<6} | {'count':^5} |")
+print('-' * SEP_NUM)
+
+people = [
```
Comment (antohaUa) [inline Lesson03_Task01.py:31]: this variable better to place right after constant and only then place code with prints

File: Lesson03_Task01.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+flat_list = list(itertools.chain.from_iterable(updated_people))
+male_count = flat_list.count('male')
+female_count = flat_list.count('female')
+
+print(f"|{'female':^8}|{female_count:^7}|")
+print(f"|{'male':^8}|{male_count:^7}|")
+print('-' * SEP_NUM)
+
+print('updated_people=')
+for person in updated_people:
```
Comment (antohaUa) [inline Lesson03_Task01.py:85]: Possible alternative print(*updated_people, sep='\n')

**Review by antohaUa — APPROVED** [review]

File: Lesson03_Task01.py [hunk 4]
```diff
@@ -0,0 +1,86 @@
+"""
+Lesson02 Task02.
```
[the remark this answers is not in the document]
Comment (ann-kolosova) [reply Lesson03_Task01.py]: fixed

File: Lesson03_Task01.py (hunk 1 above)
Comment (ann-kolosova) [reply Lesson03_Task01.py]: fixed

File: Lesson03_Task01.py (hunk 2 above)
Comment (ann-kolosova) [reply Lesson03_Task01.py:28]: fixed

File: Lesson03_Task01.py (hunk 3 above)
Comment (ann-kolosova) [reply Lesson03_Task01.py]: fixed

File: Lesson03_Task01.py [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+flat_list = list(itertools.chain.from_iterable(people))
+male_count = flat_list.count('male')
+female_count = flat_list.count('female')
+
+print(f"|{'female':^8}|{female_count:^7}|")
+print(f"|{'male':^8}|{male_count:^7}|")
+print('-' * SEP_NUM)
+
+print('people:')
```
Comment (antohaUa) [inline Lesson03_Task01.py:86]: Could be in one line:
print('people:', *people, sep='\n')

File: Lesson02_Task02.py [hunk 6]
```diff
@@ -0,0 +1,53 @@
+"""
```
Comment (antohaUa) [inline Lesson02_Task02.py:1]: For the next time please do not mix all files within pull request. Better to add one certain hometask as separate PR. 
Ex. HomeTask2 -> pull request #2, HomeTask3 -> pull request #3  ...

**Review by antohaUa — APPROVED** [review]

**Pull request merged**